subserver_permissions.go

```diff
@@ -148,3 +210,83 @@ func isLitURI(uri string) bool {
 	_, ok := litPermissions[uri]
 	return ok
 }
+
+func init() {
```

Comment:
Hmm, this is all becoming very hard to read and to follow what happens when...

Might be a slight refactor, but how about we create a new `PermissionManager` struct that runs this code during `NewPermissionManager` and keeps a copy of the full permission set in memory. Then we can call `GetPermissions(readOnly, lndBuildTags)` to get a filtered version.
And we could give it the lnd build tags later, which causes it to become "ready" (call this the "runtime permissions"). 
That way we can create a single instance of the manager before we create the `rpcProxy` and hand an instance to it (instead of a function). Then the rpcProxy would always call the `GetRuntimePerms()` which would just return an empty map while it's not "ready".

So only state and locks in the manager, everything else should become way simpler.
What do you think?

Reply:
great idea :) thanks!